backend/app/services/knowledge/knowledge_service.py

```
@@ -2193,3 +2193,116 @@ def get_table_document_by_id(
+        Raises:
+            ValueError: If validation fails or permission denied
+        """
+        from sqlalchemy.orm.attributes import flag_modified
+
+        # Get the knowledge base
+        kb = (
+            db.query(Kind)
+            .filter(
+                Kind.id == knowledge_base_id,
+                Kind.kind == "KnowledgeBase",
+                Kind.is_active == True,
```

Ruff (0.15.7)

[error] Avoid equality comparisons to `True`; use `Kind.is_active:` for truth checks

Replace with `Kind.is_active`

(E712)